Client.markRead requires the current user to be in channel members array

## What did you do?
I query channel for messages:
`channel.query(messagesPagination: pagination,
                      membersPagination: [.limit(1)],
                      watchersPagination: [.limit(1)],
                      options: [.state])`
And then call `channel.markRead()`

## What did you expect to happen?
Channel marks as read

## What happened instead?
Nothing. `Endpoint.markRead` not called

## GetStream Environment
**GetStream Chat version:**2.2.1
**GetStream Chat frameworks:** StreamChatClient, StreamChatCore

## Additional context
I looked Channel code and noticed that `readEventsEnabled` checks the current user to be in `members` array:
`public var readEventsEnabled: Bool { config.readEventsEnabled && members.contains(Member.current) }`
But I don't query members since I don't need them. I tried to query only current user with pagination `[.limit(1), .lessThanOrEqual(currentUserId)]`, but it returns error
```
[22 May 16:22:21.346] [INFO] 📦 Request Body {
  "messages" : {
    "limit" : 25
  },
  "state" : true,
  "watchers" : {
    "limit" : 1
  },
  "members" : {
    "limit" : 1,
    "id_lte" : "65"
  }
}
[22 May 16:22:21.512] [DEBUG] ⬅️ Response 400 (165 bytes): https://chat-proxy-us-east.stream-io-api.com/channels/messaging/482/query?api_key=twj335e5wcah&connection_id=200fe970-8e38-4ade-bc0d-ad950668d6ee
[22 May 16:22:21.512] [ERROR] 📦 {
  "code" : 4,
  "message" : "GetOrCreateChannel failed with error: \"expected number for field \"members.id_lte\" but got number `0`\"",
  "StatusCode" : 400,
  "duration" : "0.00ms"
}
```
Also, docs say that we can use only `.limit` and `.offset` pagination for `membersPagination`. If I pass `[.limit(1000)]` or `[]` for `membersPagination`, `markRead` works all right, but as I understand there will be returned only default page size of members, so it's not guaranteed that the current user will appear in that page for channels with a lot of members.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 # Upcoming
 
 ### 🔄 Changed
+- Remove the need for querying members in order to mark a channel as read [#280](https://github.com/GetStream/stream-chat-swift/issues/280)
 
 # [2.2.7](https://github.com/GetStream/stream-chat-swift/releases/tag/2.2.7)
 _July 29, 2020_
@@ -27,7 +28,6 @@ _July 29, 2020_
 - All iOS versions will use Starscream as default websocket provider until native provider issue is resolved. See [#315](https://github.com/GetStream/stream-chat-swift/issues/315). [#357](https://github.com/GetStream/stream-chat-swift/issues/357)
 
 ### 🐞 Fixed
-- Remove need for member query in order to mark channel as read. [#280](https://github.com/GetStream/stream-chat-swift/issues/280)
 - Reintroduce Hashable conformances removed in 2.2.1 [#368](https://github.com/GetStream/stream-chat-swift/pull/368)
 - `ChannelPresenter.lastMessage` not updated on message edited or deleted [#351](https://github.com/GetStream/stream-chat-swift/issues/351)
 - Link tap in messages sometimes not detected or detected in wrong place [#350](https://github.com/GetStream/stream-chat-swift/issues/350)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ _July 29, 2020_
 - All iOS versions will use Starscream as default websocket provider until native provider issue is resolved. See [#315](https://github.com/GetStream/stream-chat-swift/issues/315). [#357](https://github.com/GetStream/stream-chat-swift/issues/357)
 
 ### 🐞 Fixed
+- Remove need for member query in order to mark channel as read. [#280](https://github.com/GetStream/stream-chat-swift/issues/280)
 - Reintroduce Hashable conformances removed in 2.2.1 [#368](https://github.com/GetStream/stream-chat-swift/pull/368)
 - `ChannelPresenter.lastMessage` not updated on message edited or deleted [#351](https://github.com/GetStream/stream-chat-swift/issues/351)
 - Link tap in messages sometimes not detected or detected in wrong place [#350](https://github.com/GetStream/stream-chat-swift/issues/350)

--- a/Sources/Client/Client/Client+Channel.swift
+++ b/Sources/Client/Client/Client+Channel.swift
@@ -299,10 +299,6 @@ public extension Client {
     ///   - completion: a completion block with `Event`.
     @discardableResult
     func markRead(channel: Channel, _ completion: @escaping Client.Completion<Event>) -> Cancellable {
-        guard channel.readEventsEnabled else {
-            return Subscription.empty
-        }
-        
         logger?.log("🎫 Mark Read")
         
         return request(endpoint: .markRead(channel)) { (result: Result<EventResponse, ClientError>) in

--- a/Sources/Client/Client/Client+Channel.swift
+++ b/Sources/Client/Client/Client+Channel.swift
@@ -296,11 +296,10 @@ public extension Client {
     /// Mark messages in the channel as read.
     /// - Parameters:
     ///   - channel: a channel.
-    ///   - skipReadEventsEnabledCheck: if true, an API call is made even if the current user is not present in the channel's member list. Default is false.
     ///   - completion: a completion block with `Event`.
     @discardableResult
-    func markRead(channel: Channel, skipReadEventsEnabledCheck: Bool = false, _ completion: @escaping Client.Completion<Event>) -> Cancellable {
-        guard skipReadEventsEnabledCheck || channel.readEventsEnabled else {
+    func markRead(channel: Channel, _ completion: @escaping Client.Completion<Event>) -> Cancellable {
+        guard channel.readEventsEnabled else {
             return Subscription.empty
         }
         

--- a/Sources/Client/Client/Client+Channel.swift
+++ b/Sources/Client/Client/Client+Channel.swift
@@ -296,9 +296,14 @@ public extension Client {
     /// Mark messages in the channel as read.
     /// - Parameters:
     ///   - channel: a channel.
+    ///   - skipReadEventsEnabledCheck: if true, an API call is made even if the current user is not present in the channel's member list. Default is false.
     ///   - completion: a completion block with `Event`.
     @discardableResult
-    func markRead(channel: Channel, _ completion: @escaping Client.Completion<Event>) -> Cancellable {
+    func markRead(channel: Channel, skipReadEventsEnabledCheck: Bool = false, _ completion: @escaping Client.Completion<Event>) -> Cancellable {
+        guard skipReadEventsEnabledCheck || channel.readEventsEnabled else {
+            return Subscription.empty
+        }
+        
         logger?.log("🎫 Mark Read")
         
         return request(endpoint: .markRead(channel)) { (result: Result<EventResponse, ClientError>) in

